smoothie.js

```diff
@@ -671,6 +671,11 @@
       var timeSeries = this.seriesSet[d].timeSeries,
           dataSet = timeSeries.data,
           seriesOptions = this.seriesSet[d].options;
+          
+      if (seriesOptions.lineWidth === 0) {
+        context.restore();
+        continue;
```

Comment:
Good catch... in fact i've already changed it a bit, as the "lineDash" property doesn't gets saved in the context.